waveflag.c

```diff
@@ -409,7 +409,12 @@ wave_flag (const char *filename, const char *out_prefix)
 	*out = '\0';
 	strcat (out, out_prefix);
         // diff from upstream. we call this a bit differently, filename might not be in cwd.
-	strcat (out, basename(filename));
+
+        // basename wants a non-const argument.  The problem here is paths that end in a
+        // slash, POSIX basename removes them while GNU just returns a pointer to that
+        // slash.  Since this is supposed to be a filename such input is illegal for us.
+        // We're already not checking for overflow of the output buffer anyway...
+	strcat (out, basename((char *) filename));
```

Comment:
We could.  Right now we just have a really large buffer (1000 bytes).  It's pretty unlikely a filepath would get this long.  Also strncat is a limit on the length of the source, not the length of the resulting string, which would be more useful, and technically we'd have to check every call to append to the buffer.  I thought about doing this but didn't, there's a lot of ways this code isn't bulletproof.